src/services/firebase/firestore.ts

```diff
@@ -225,14 +227,22 @@ export class FirestoreServices {
         .orderBy('createdAt', 'desc')
         .limit(maxPageSize)
         .get();
-      querySnapshot.forEach((doc) => {
```

Comment:
The forEach loop does not wait for the await statements within it to complete before continuing.
To handle this correctly, use a for...of loop to ensure that each asynchronous operation is awaited before moving to the next iteration